src/Executor/Executor.php

```diff
@@ -430,7 +430,7 @@ private static function completeField(ExecutionContext $exeContext, Type $fieldT
         if ($fieldType instanceof ListOfType) {
             $itemType = $fieldType->getWrappedType();
             Utils::invariant(
-                is_array($result) || $result instanceof \ArrayObject,
+                is_array($result) || $result instanceof \ArrayAccess,
```

Comment:
Technically, we need to check for `Traversable` interface here, not `ArrayAccess`, because all we do later is iterating `$result` in foreach loop.

Will merge this pull request if you change it to `Traversable`.